homeassistant/components/wemo/__init__.py

```diff
@@ -217,6 +213,21 @@ def async_stop_discovery(self) -> None:
             self._stop()
             self._stop = None
 
+    async def discover_statics(self):
+        """Initialize or Re-Initialize connections to statically configured devices."""
+        if self._static_config:
+            _LOGGER.debug("Adding statically configured WeMo devices")
+            for device in await asyncio.gather(
+                *[
+                    self._hass.async_add_executor_job(
+                        validate_static_config, host, port
```

Comment:
Please create a function that validates all the devices and returns a list of validated devices, that we can schedule once on the executor. We don't want to consume threads greedily from the pool.

Reply:
There is this too, which limits the number of threads.
https://github.com/home-assistant/core/pull/46821/files#diff-529af4cf0518764fef83c0d7357f912ab8d3a264da718b6c0e83fee9b644f5a5R121

Reply:
@MartinHjelmare would implementing the max concurrency approach from this other PR address your concern? I imagine it's better to be consistent between static and auto-discovered devices, but if there's a strong preference for only scheduling one task, I can do it that way instead.

Reply:
It's ok if there's a reason to do it concurrent.

Reply:
Ok, I've added the max concurrency stuff from that other PR